staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter.go

```diff
@@ -111,22 +111,24 @@ func (cfgCtlr *configController) Handle(ctx context.Context, requestDigest Reque
 		return
 	}
 	klog.V(7).Infof("Handle(%#+v) => fsName=%q, distMethod=%#+v, plName=%q, isExempt=%v, queued=%v", requestDigest, fs.Name, fs.Spec.DistinguisherMethod, pl.Name, isExempt, queued)
-	var executed bool
-	idle := req.Finish(func() {
+	var executed, idle bool
+	defer func() {
+		klog.V(7).Infof("Handle(%#+v) => fsName=%q, distMethod=%#+v, plName=%q, isExempt=%v, queued=%v, Finish() => idle=%v", requestDigest, fs.Name, fs.Spec.DistinguisherMethod, pl.Name, isExempt, queued, idle)
+		cfgCtlr.maybeReap(pl.Name)
```

Comment:
It is not functionally incorrect to call `cfgCtlr.maybeReap` in all cases, but it is more expensive than necessary.  If the `idle` variable were simply initialized to `true` rather than `false` then the invocation of `maybeReap` could be conditional on `idle`.